packages/Swiper/src/index.tsx

```diff
@@ -1,175 +1,181 @@
+      'aria-roledescription': 'slide',
+      'aria-label': `${i + 1} of ${Children.toArray(children).length}`,
+    })
+  })
+  const slidesLength = slides.length
 
-  // Get array with indexes of visible slides so we know which ones are (aria-)hidden
-  const visibleSlides = Array(slidesToShow)
-    .fill('')
-    .map((_, idx) => pageIdx + idx)
+  const numberOfPage = Math.ceil(slidesLength / currentSlidesPerView)
+  const bullets = Array.from(Array(numberOfPage).keys())
+  const isFirstPage = currentPage === 0
+  const isLastPage = currentPage === bullets.length - 1
 
-  return (
-    <S.Wrapper
-      {...rest}
-      aria-live="off"
-      aria-roledescription="carousel"
-      id={id}
-      ref={ref}
-      role="region"
-    >
-      <S.Swiper slidesToShow={slidesToShow} translateX={translateX}>
-        {children.map((child, idx) =>
-          cloneElement(child as React.ReactElement, {
-            // eslint-disable-next-line react/no-array-index-key
-            key: idx,
-            id: `${id}-${idx}`,
-            role: 'tabpanel',
-            'aria-hidden': !visibleSlides.includes(idx),
-            'aria-roledescription': 'slide',
-            'aria-label': `${idx + 1} of ${numberOfSlides}`,
-          })
-        )}
-      </S.Swiper>
-      <S.Pagination
-        className="swiper-pagination"
-        data-testid={dataTestId && `${dataTestId}-pagination`}
-        role="tablist"
-      >
-        {children.map((_, idx) => {
-          const props: RenderPaginationProps = {
-            idx,
-            role: 'tab',
-            'aria-controls': `${id}-${idx}`,
-            'aria-label': `${idx + 1} of ${numberOfSlides}`,
-            'aria-selected': idx === pageIdx,
-            onClick: () => setPageIdx(idx),
-            pageIdx,
-          }
-          if (renderPaginationItem) {
-            return renderPaginationItem(props)
-          }
-          // eslint-disable-next-line react/no-array-index-key
-          return <S.Bullet active={idx === pageIdx} key={idx} {...props} />
-        })}
-      </S.Pagination>
-      {prevButton && (
-        <S.Prev
-          aria-controls={id}
-          aria-label="Previous slide"
-          className="swiper-prev"
-          data-testid={dataTestId && `${dataTestId}-button-prev`}
-          disabled={!loop && pageIdx === 0}
-          onClick={goPrev}
-        >
-          {prevButton}
-        </S.Prev>
-      )}
-      {nextButton && (
-        <S.Next
-          aria-controls={id}
-          aria-label="Next slide"
-          className="swiper-next"
-          data-testid={dataTestId && `${dataTestId}-button-next`}
-          disabled={!loop && pageIdx >= numberOfSlides - slidesToShow}
-          onClick={goNext}
-        >
-          {nextButton}
-        </S.Next>
-      )}
-    </S.Wrapper>
-  )
-})
+  const firstPageToShow = centeredSlides
+    ? // if centeredSlides is true, we calculate which number is the middle page
+      Math.ceil(numberOfPage / 2)
```

Comment:
Thanks for the explanation, it's much clearer!
Maybe we could say it in the doc that the firstSlideToShow and centeredSlides props cannot be combined?
I don't know if you've already written it

Reply:
Yes you're right, I just add the info on the doc that `firstSlideToShow` won't be used if centeredSlides is true

Reply:
That's perfect, thanks 🔥